nirc_ehr/resources/queries/study/deaths.js

```diff
@@ -62,11 +78,22 @@ function onUpsert(helper, scriptErrors, row, oldRow) {
             if (idMap[row.Id].calculated_status.toUpperCase() === 'DEAD' && row.QCStateLabel.toUpperCase() === 'COMPLETED') {
                 EHR.Server.Utils.addError(scriptErrors, 'Id', 'Death record already exists for this animal.', 'ERROR');
             }
+
             // check if the animal is at the center
             if (idMap[row.Id].calculated_status.toUpperCase() === 'SHIPPED') {
                 EHR.Server.Utils.addError(scriptErrors, 'Id', 'Animal is not at the center.', 'ERROR');
             }
 
+            // check if the animal is pending review on Death record
+            // Note: Upon 'Submit Death', the QCState will get set to 'REQUEST: PENDING', and upon 'Submit for Review',
+            // the QCState will get set to 'Review Required' - this way we can distinguish between the two states in the Death/Necropsy workflow. And if the user
+            // tries to submit a Death record for an animal that already exists with one of these two QCStates, below error message will be displayed.
+            if (row.QCStateLabel.toUpperCase() === 'IN PROGRESS' &&
```

Comment:
I'm not totally following why IN PROGRESS is being used here. Can you walk me through that?

Reply:
Comment updated.